packages/trpc/server/routers/viewer.tsx

```diff
@@ -708,6 +709,25 @@ const loggedInViewerRouter = router({
           },
         });
       }
+      // Revalidate booking pages
+      const res = ctx.res as NextApiResponse;
+      if (typeof res?.revalidate !== "undefined") {
+        const eventTypes = await prisma.eventType.findMany({
+          where: {
+            userId: user.id,
+            team: null,
+            hidden: false,
+          },
+          select: {
+            id: true,
+            slug: true,
+          },
+        });
+        // waiting for this isn't needed
+        Promise.all(eventTypes.map((eventType) => res?.revalidate(`/${ctx.user.username}/${eventType.slug}`)))
```

Comment:
This can take some time theoretically. Is it possible that the serverless function stops execution midway because the response has already been sent?

Maybe we should have a before and after log to track that this works successfully everytime(I see an "after" log, we can just add a "before" log)

Also, we can have a whitelist of changes that impact booking pages to probably avoid, unnecessary revalidate calls in the system e.g. Password, session timeout and others' don't impact booking pages.